v-next/hardhat/src/internal/builtin-global-options.ts

```diff
@@ -0,0 +1,56 @@
+import type { GlobalOptionDefinitions } from "../types/global-options.js";
+
+import { globalOption, ArgumentType } from "../config.js";
+
+export const BUILTIN_GLOBAL_OPTIONS_DEFINITIONS: GlobalOptionDefinitions =
+  new Map([
+    [
+      "config",
+      {
+        pluginId: "hardhat",
```

Comment:
Should we make this optional? Or maybe switch to `builtin` or something more explicit?

Reply:
I thought about making it optional, but this required less changes. `builtin` also works for me. what do you prefer?

Reply:
Done in https://github.com/NomicFoundation/hardhat/pull/5494/commits/26c1d6c8beb6cca12942f03108e0029d466fefe8